src/Symfony/Component/Config/ResourceCheckerConfigCache.php

```diff
@@ -185,4 +179,20 @@ public static function handleUnserializeCallback($class)
     {
         trigger_error('Class not found: '.$class);
     }
+
+    /**
+     * Returns the associated resources that are monitored by this cache.
+     */
+    public function getResources(): ?iterable
+    {
+        $metadata = $this->getMetaFile();
+
+        if (!is_file($metadata)) {
+            return null;
+        }
+
+        $meta = $this->safelyUnserialize($metadata);
+
+        return $meta ?: null;
```

Comment:
I think this is gonna be called twice, isn't it? once for resource checking, twice when MessageCatalogue::getResources() is called? should then be cached if confirmed?

Reply:
I just saw that as well. We can heavily simplify the implementation then. 👍